[MSTA-681] Unify monitor group retention docs and add DGR preview panel (#37884)
DataDog/documentation: content/en/monitors/configuration/_index.md

For the multi alert monitor types that support custom retention, you can set a value between 1 hour and 72 hours. Select {{< ui >}}Remove the non-reporting group after N (length of time){{< /ui >}}.

{{< img src="/monitors/create/group_retention_time.png" alt="Group Retention Time Option" style="width:70%;">}}

Similar to the [Auto-resolve option][3], group retention works when data is no longer being submitted. The start time of the group retention and the Auto-resolve option are **identical** as soon as the monitor query returns no data.

Some use cases to define a custom group retention time include:

- When you would like to drop the group immediately or shortly after data stops reporting
- When you would like to keep the group in the status for as long as you usually take for troubleshooting

**Note**: The custom group retention time option requires a multi alert monitor that supports the [`On missing data`][4] option. These monitor types are APM Trace Analytics, Audit Logs, CI Pipelines, Error Tracking, Events, Logs, and RUM monitors.

> <div class="alert alert-info"><strong>Preview: Dynamic group retention</strong><p>For monitors with a large number of groups, Datadog automatically shortens how long a group is kept after it stops reporting data: the more frequently groups appear and disappear, the shorter this period becomes, keeping the monitor fast and efficient. This only affects groups that have gone silent; groups actively reporting data are never removed.</p></div>

Reviewer: ```suggestion
<div class="alert alert-info"><strong>Preview: Dynamic group retention</strong><p>Dynamic group retention is in Preview and applies to newly created monitors. For monitors with a large number of groups, Datadog automatically shortens how long a group is kept after it stops reporting data: the more frequently groups appear and disappear, the shorter this period becomes, keeping the monitor fast and efficient. This only affects groups that have gone silent; groups actively reporting data are never removed.</p></div>
```

> <div class="alert alert-info"><strong>Preview: Dynamic group retention</strong><p>Dynamic group retention is in Preview and applies to newly created monitors. For monitors with a large number of groups, Datadog automatically shortens how long a group is kept after it stops reporting data: the more frequently groups appear and disappear, the shorter this period becomes, keeping the monitor fast and efficient. This only affects groups that have gone silent; groups actively reporting data are never removed.</p></div>

Reviewer: ```suggestion
<div class="alert alert-info"><strong>Preview: Dynamic group retention</strong><p>Dynamic group retention is in Preview and applies to newly created monitors. For monitors with a large number of groups, Datadog automatically shortens how long a group is kept after it stops reporting data. The more frequently groups appear and disappear, the shorter this period becomes, keeping the monitor fast and efficient. This only affects groups that have gone silent; groups actively reporting data are never removed.</p></div>
```

#### Group retention time

Group retention controls how long a monitor group is kept in the monitor's status after data stops reporting. After the retention period passes, the group ages out and is dropped from the monitor status.

By default, a group keeps the status for 24 hours before it is dropped. Host monitors and service checks that notify on missing data keep the status for 48 hours.

> For the multi alert monitor types that support it, you can set a custom retention between 1 hour and 72 hours. Select **Remove the non-reporting group after `N (length of time)`**.

Reviewer: ```suggestion
For the multi alert monitor types that support custom retention, you can set a value between 1 hour and 72 hours. Select {{< ui >}}Remove the non-reporting group after N (length of time){{< /ui >}}.
```

{{< img src="/monitors/create/group_retention_time.png" alt="Group Retention Time Option" style="width:70%;">}}

Similar to the [Auto-resolve option][3], group retention works when data is no longer being submitted. The start time of the group retention and the Auto-resolve option are **identical** as soon as the monitor query returns no data.

Some use cases to define a custom group retention time include:

- When you would like to drop the group immediately or shortly after data stops reporting
- When you would like to keep the group in the status for as long as you usually take for troubleshooting

**Note**: The custom group retention time option requires a multi alert monitor that supports the [`On missing data`][4] option. These monitor types are APM Trace Analytics, Audit Logs, CI Pipelines, Error Tracking, Events, Logs, and RUM monitors.

> {{< notice label="PREVIEW" color="#9364cd" >}}

Reviewer: Can we replace with an [existing shortcode](https://datadoghq.atlassian.net/wiki/spaces/docs4docs/pages/1182958054/Markdown+Shortcode+Reference#Alert-boxes)?
